src/tests/test_aead.cpp

```diff
@@ -61,12 +67,20 @@ class AEAD_Tests final : public Text_Based_Test {
          result.test_eq("key is set", enc->has_keying_material(), true);
 
          if(is_siv == false) {
-            result.test_throws("Cannot process data until nonce is set (enc)", [&]() { enc->update(garbage); });
-            result.test_throws("Cannot process data until nonce is set (enc)", [&]() { enc->finish(garbage); });
+            result.test_throws("Cannot process data until nonce is set (enc)", [&]() {
+               auto garbage = get_garbage();
+               enc->update(garbage);
+            });
+            result.test_throws("Cannot process data until nonce is set (enc)", [&]() {
+               auto garbage = get_garbage();
+               enc->finish(garbage);
```

Comment:
With #4880 this would re-alloc the garbage already in the base class (`Cipher_Mode`) to make room for the to-be-calculated tag. Then, the concrete implementation throws because of the (expected) invalid state.